packages/babel-plugin-transform-destructuring/test/api.js

```diff
@@ -0,0 +1,70 @@
+import { parse } from "@babel/parser";
+import { transformFromAstSync } from "@babel/core";
+import _traverse from "@babel/traverse";
+const traverse = _traverse.default;
+
+import path from "path";
+import { fileURLToPath } from "url";
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+import glob from "glob";
+import { promises as fs } from "fs";
+
+const files = await new Promise((resolve, reject) => {
+  glob(
+    path.resolve(__dirname, "fixtures/**/@(exec|input).js"),
+    function (err, files) {
+      if (err) reject(err);
+      else resolve(files);
+    },
+  );
+});
+
+describe("regression-14438", function () {
+  describe.each(files)("fixture: %s", file => {
+    let src, originalAst, transformed;
+
+    beforeEach(async () => {
+      src = await fs.readFile(file, "utf-8");
+
+      originalAst = parse(src, {
+        allowImportExportEverywhere: true,
+        allowAwaitOutsideFunction: true,
+        allowReturnOutsideFunction: true,
+        allowSuperOutsideMethod: true,
+        allowUndeclaredExports: true,
+        errorRecovery: true,
+        attachComment: false,
+        createParenthesizedExpressions: true,
+      });
+
+      transformed = transformFromAstSync(originalAst, src, {
+        plugins: [
+          [
+            "@babel/plugin-transform-destructuring",
+            { useBuiltIns: true, loose: true },
+          ],
+        ],
+        configFile: false,
+        ast: true,
+        code: true,
+      });
+    });
+
+    it("has correct scope information after transform", function () {
+      if (transformed.ast) {
+        // exclude transformation failures: they're tested by the other cases
+        traverse(transformed.ast, {
+          VariableDeclarator(path) {
+            const b = path.scope.getBinding(path.get("id").node.name);
+            if (!b) {
```

Comment:
This branch is actually unreachable because `traverse` will initialize the scope info from the _transformed_ ast.

To test whether the scope info has been updated, the visitor has to be merged with the destructuring plugin visitor. We can wrap it into a new plugin:

```js
// checkScopeInfo.js
module.exports = (babel) => {
  return {
    Program: {
      exit(programPath) {
        // so that this plugin is always called after the transform-destructuring plugin
        programPath.traverse(checkScopeVisitor)
      }
    }
  }
}
```

Reply:
> This branch is actually unreachable

I'm unsure about this, `git checkout master -- src` to undo all additions to `src` and ` TEST_GREP="regression-14438" make test-only` leaves 27 tests failing

> To test whether the scope info has been updated, the visitor has to be merged with the destructuring plugin visitor. We can wrap it into a new plugin

I don't understand this suggestion, perhaps due to my unfamiliarity with babel source. Are you suggesting doing for all transform plugins what I've done here for `plugin-transform-destructuring`?

I should mention that I think generally scope information would be added in plugins because people would use `path.scope.registerDeclaration`. But since in this plugin the ast nodes are created all at once and then appended all at once, there's an issue. 

I'll take care of all the `path.scope => scope` comments. Thanks for the review :)